src/org/jointheleague/level4/minesweeper/MineSweeper.java

```diff
@@ -0,0 +1,132 @@
+package org.jointheleague.level4.minesweeper;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.GridLayout;
+import java.util.Optional;
+import java.util.Random;
+import java.util.stream.IntStream;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+/**
+ * Implement the Mine Sweeper game by following the instructions in the
+ * comments. All instructions are marked with "TODO".
+ */
+public class MineSweeper {
+    private static final int WIDTH = 10;
+    private static final int HEIGHT = 15;
+    private static final int CELL_SIZE = 20;
+    private static final int NUM_MINES = 15;
+    
+    final JFrame frame = new JFrame();
+    private final JButton[][] buttons = new JButton[HEIGHT][WIDTH];
+    private final Random rng = new Random();
+    private Optional<boolean[][]> mines = Optional.empty();
+    private int numCellsRemaining;
+    
+    /**
+     * Initializes:
+     * 1. The `mines` variable with `NUM_MINES` randomly distributed mines
+     * 2. The `numCellsRemaining` variable to the number of non-mine
+     *    cells (`WIDTH` * `HEIGHT` - `NUM_MINES`).
+     * 
+     * @param firstCellX X-index of first cell opened. This cannot be a mine.
```

Comment:
I usually avoid `x` and `y` and use `row` and `column` (or `row` and `col`, or `r` and `c`). This avoids the confusion about whether rows corresponds to `y` or `x`.